libs/wingc/src/jsify.rs

```diff
@@ -710,6 +695,315 @@ impl JSifier {
+		let mut lines = vec![];
+		lines.push("{".to_string());
+
+		// If there's no parent then this resource id derived from the base resource class (core.Resource) and we need
+		// to manually call its super
+		if no_parent {
+			lines.push("	super(scope, id);".to_string());
+		}
+
+		for statement in constructor.statements.statements.iter() {
+			let statement_str = self.jsify_statement(
+				constructor.statements.env.borrow().as_ref().unwrap(),
+				statement,
+				Phase::Preflight,
+			);
+			let result = statement_str.split("\n");
+			for l in result {
+				lines.push(format!("  {}", l));
+			}
+		}
+
+		lines.push("}".to_string());
+
+		format!(
+			"constructor(scope, id, {}) {{\n{}\n}}",
+			constructor
+				.parameters
+				.iter()
+				.map(|(name, _)| name.name.clone())
+				.collect::<Vec<_>>()
+				.join(", "),
+			lines.join("\n")
+		)
+	}
+
+	fn jsify_toinflight_method(
+		&self,
+		resource_name: &Symbol,
+		captured_fields: &[(String, TypeRef, Vec<String>)],
+	) -> String {
+		let inner_clients = captured_fields
+			.iter()
+			.map(|(inner_member_name, _, _)| {
+				format!(
+					"const {}_client = this.{}._toInflight();",
+					inner_member_name, inner_member_name,
+				)
+			})
+			.collect::<Vec<_>>();
+
+		let client_relative_path = format!("clients/{}.inflight.js", resource_name.name);
+		format!(
+			"_toInflight() {{\n{}\nconst self_client_path = {};\nreturn core.NodeJsCode.fromInline(`(new (require(\"${{self_client_path}}\")).{}_inflight({{{}}}))`);\n}}",
+			inner_clients.join("\n"),
+			Self::js_resolve_file(&client_relative_path),
+			resource_name.name,
+			captured_fields
+				.iter()
+				.map(|(inner_member_name, _, _)| format!("{}: ${{{}_client.text}}", inner_member_name, inner_member_name))
+				.join(", ")
+		)
+	}
+
+	// Write a client class for a file for the given resource
+	fn jsify_resource_client(
+		&self,
+		env: &SymbolEnv,
+		name: &Symbol,
+		captured_fields: &[(String, TypeRef, Vec<String>)],
+		inflight_methods: &[&(Symbol, FunctionDefinition)],
+		parent: &Option<Type>,
+	) {
+		// Handle parent class: Need to call super and pass its captured fields (we assume the parent client is already written)
+		let mut parent_captures = vec![];
+		if let Some(Type::CustomType(parent)) = parent {
+			let parent_type = resolve_custom_type(parent, env, 0).unwrap();
+			parent_captures.extend(self.get_captures(parent_type));
+		}
+
+		// Get the fields that are captured by this resource but not by its parent, they will be initialized in the generated constructor
+		let my_captures = captured_fields
+			.iter()
+			.filter(|(name, _, _)| !parent_captures.iter().any(|(n, _, _)| n == name))
+			.collect_vec();
+
+		let super_call = if parent.is_some() {
+			format!(
+				"  super({});",
+				parent_captures
+					.iter()
+					.map(|(name, _, _)| name.clone())
+					.collect_vec()
+					.join(", ")
+			)
+		} else {
+			"".to_string()
+		};
+
+		// TDOD jsify inflight fields
```

Comment:
Tracking issue 🙏🫶

Reply:
@yoav-steinberg - perhaps it's enough to add a comment in #864

Reply:
done